llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -1610,6 +1610,10 @@ void NVPTXAsmPrinter::emitFunctionParamList(const Function *F, raw_ostream &O) {
             }
             Align ParamAlign = I->getParamAlign().valueOrOne();
             O << ".align " << ParamAlign.value() << " ";
+          } else if (I->getParamAlign().valueOrOne() != 1) {
```

Comment:
We need a comment somewhere under `if (PTy)` explaining that CUDA kernels assume that pointers are in global AS, with the pointer to the docs. Otherwise it's confusing to see AS-specific pointer annotations above, but not here.

Reply:
Added the comment. Thanks.